src/contextprotector/mcp_json_cli.py

```diff
@@ -0,0 +1,977 @@
+        if new_json.strip() == self.original_json.strip():
+            print("No changes to save.")
+            return True
+
+        # Show the diff
+        print("\n" + "=" * 60)
+        print("PROPOSED CHANGES")
+        print("=" * 60)
+
+        original_lines = self.original_json.splitlines(keepends=True)
+        new_lines = new_json.splitlines(keepends=True)
+
+        diff = difflib.unified_diff(
+            original_lines,
+            new_lines,
+            fromfile=f"{self.file_path} (original)",
+            tofile=f"{self.file_path} (modified)",
+            lineterm="",
+        )
+
+        for line in diff:
+            if line.startswith("+++") or line.startswith("---"):
+                print(f"\033[1m{line}\033[0m", end="")  # Bold
+            elif line.startswith("@@"):
+                print(f"\033[36m{line}\033[0m", end="")  # Cyan
+            elif line.startswith("+"):
+                print(f"\033[32m{line}\033[0m", end="")  # Green
+            elif line.startswith("-"):
+                print(f"\033[31m{line}\033[0m", end="")  # Red
+            else:
+                print(line, end="")
+
+        print("\n" + "=" * 60)
+
+        # Ask for confirmation
+        while True:
+            confirm = input("Do you want to save these changes? [y/N]: ").strip().lower()
+            if confirm in ("", "n", "no"):
+                print("Changes not saved.")
+                return False
+            elif confirm in ("y", "yes"):
+                break
+            else:
+                print("Please enter 'y' for yes or 'n' for no.")
+
+        # Create backup if file exists
+        if self.file_path.exists():
+            backup_path = self.file_path.with_suffix(self.file_path.suffix + ".backup")
+            backup_path.write_text(self.original_json)
+            print(f"Backup saved to: {backup_path}")
+
+        # Save the file
+        try:
+            if hasattr(self.config, "raw_data"):
+                # Unified config - has save() method
+                self.config.save()
+            else:
+                # Legacy config - save with file path
+                self.config.save()
+            self.original_json = new_json  # Update our reference
+            return True
+        except Exception as e:
+            print(f"Error saving file: {e}")
+            return False
+
+
+def manage_mcp_json_file(file_path: str) -> None:
+    """Manage MCP JSON file management interactively.
+
+    Args:
+    ----
+        file_path: Path to the MCP JSON configuration file
+
+    """
+    manager = MCPJsonManager(file_path)
+    manager.run()
+
+
+class AllMCPJsonManager:
+    """Manager for discovering and selecting from all available MCP JSON configuration files."""
+
+    def __init__(self) -> None:
+        """Initialize the all MCP JSON manager."""
+        # (client_name, path, server_count, environment)
+        self.discovered_configs: list[tuple[str, str, int, str | None]] = []
```

Comment:
not sure it makes sense to abstract config elements in a separate definition, e.g.
```
from dataclasses import dataclass

  @dataclass
  class DiscoveredConfig:
      client_name: str
      config_path: str
      server_count: int
      environment: str | None = None

      @property
      def is_parseable(self) -> bool:
          return self.server_count != -1

      @property
      def has_servers(self) -> bool:
          return self.server_count > 0
```
Feels that the comment would want that 😄